src/utils/forms/FormFactory.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import {
+    FormControl,
+    FormGroup,
+    Col,
+    ControlLabel,
+    HelpBlock,
+    ButtonGroup,
+    DropdownButton,
+    MenuItem,
+    Checkbox
+  } from 'patternfly-react';
+
+export const FormFactory = (props) => {
+  return Object.keys(props.fields).map(key => {
+    if (props.fields[key].isVisible && props.fields[key].isVisible(props.fieldsValues) === false){
+      return undefined;
```

Comment:
just `return;` is enough

Reply:
this results in eslint warning. Anyways I avoided it by filtering the fields which are not visible and than map only the visible fields.